### DISCUSSION TITLE
ECONNREFUSED error for Angular app when making a request outside of app first.

### DISCUSSION THREAD

**Discussion by ChristianPavilonis** [cypress-io/cypress#25301] [Questions and Help]
I'm encountering strange behavior for my tests when my test makes a request to an API to setup some data first before `cy.visit()` is called.

My project has an angular frontend with an express api, to reset my data and login before tests I make a request to the express backend using `cy.request('localhost:3000/...` then I call `cy.visit('/');` I get the `ECONNREFUSED` error. And I have to kill cypress and start over.

But, if I comment out my `cy.request()` method first, run the tests, they will visit correctly, then I can un-comment out the code and all my tests will work.

Example:

This will create the error:
```js
describe('foo', () => {
  beforeEach(() => {
    cy.refreshDatabase();
    cy.visit('/foo');
  });

  ...
});
```

This fixes the error, but I must first exit cypress, reopen it, remove the code, run it once, finally add the code back, then I can work.
```js
describe('foo', () => {
  beforeEach(() => {
    // cy.refreshDatabase();
    cy.visit('/foo');
  });

  ...
});
```

The `refreshDatabase` command looks like:
```js
Cypress.Commands.add('refreshDatabase', () => {
    return cy.request({
        url: 'http://localhost:3000/__cypress__/refresh-database',
        method: 'POST',
        body: {},
    }).its('body', {log: false});
});
```

Any ideas?

**Comment by AGuyCoding** [comment]
did you ever found a solution to this?
